content/en/hosting/cht/migration/preparing-for-5.md

```diff
@@ -0,0 +1,77 @@
+
+Kubernetes based deployments will need to follow the migration steps from the deprecated [CHT 4.x Helm Charts repository](https://github.com/medic/helm-charts/) to the new [CHT 5.x Helm Charts](https://github.com/medic/cht-core/tree/master/scripts/build/helm) in the main CHT repository.
+
+Please see the technical guide on how to migrate Kubernetes based deployments to 5.0.
+
+Background information:
+* [Reduce disk space with CouchDB Nouveau (TCO)](https://github.com/medic/cht-core/issues/9542)
+
+## Upgrade service removed from Kubernetes
+
+{{< callout type="info" >}} Applies to: [Kubernetes](http://localhost:1313/hosting/cht/kubernetes/) hosted CHT instances {{< /callout >}}
+
+In CHT 4.x there are two ways to upgrade a Kubernetes hosted CHT instance:
+
+1. Click "Stage" and then "Upgrade" in the CHT administration interface.  
+2. Push a new helm chart to Kubernetes that specified different versions.
+
+A recurring problem seen on production deployments was that, over time, CHT administrators would use the CHT administration interface to upgrade.  Then, during a migration or a restore from backup, Kubernetes administrators would inadvertently downgrade a CHT instance by using an out of date Helm chart.
+
+To avoid this situation, the upgrade service has been removed from Kubernetes based deployments.
+
+Kubernetes based deployments will need to follow the migration steps from the deprecated [CHT 4.x Helm Charts repository](https://github.com/medic/helm-charts/) to the new [CHT 5.x Helm Charts](https://github.com/medic/cht-core/tree/master/scripts/build/helm) in the main CHT repository. Please see the TK technical guide on how to migrate Kubernetes based deployments to 5.0.
+
+Background information:
+* [Hide upgrade button in admin app for k8s deployments, while still allowing staging upgrades](https://github.com/medic/cht-core/issues/9954)
+* [Remove upgrade service from Helm charts](https://github.com/medic/cht-core/issues/10186)
+
+## Token login requires `app_url`
+
+{{< callout type="info" >}} Applies to: Instances that use [token login](/building/login/#magic-links-for-logging-in-token-login) {{< /callout >}}
+
+For instances that use token login,  be sure declare `app_url` in your [settings file](/building/reference/app-settings/token_login/#app_settingsjson-token_login). This is backwards compatible and safe to do while still on CHT 4.x.  
+
+Background information:
+* [Require `app_url` to be set when enabling `token_login`](https://github.com/medic/cht-core/issues/9983)
+
+## Enabling languages via generated docs
+
+{{< callout type="info" >}} Applies to: Instances that enable languages via web interface {{< /callout >}}
+
+As of CHT 4.20, deployment administrators can enable or disable supported languages [via the settings file](/building/reference/app-settings/#app_settingsjson). CHT 5.0 removes this web interface to enable or disable languages and requires this to be done via App Settings.
```

Comment:
> As of CHT 4.20

Is this note about `4.20` correct?  I don't see anything in the [release notes](https://docs.communityhealthtoolkit.org/releases/4_20_0) about languages.

Reply:
D'oh!  I miss-read that release - thanks for double checking.  It's `4.2.0` and it IS [mentioned there](https://docs.communityhealthtoolkit.org/releases/4_2_0/) :sweat_smile: 

fixed in 91e494c2